src/instancemanager.h

```diff
@@ -24,49 +24,91 @@ along with Mod Organizer.  If not, see <http://www.gnu.org/licenses/>.
 #include <QString>
 #include <QSettings>
 
+namespace MOBase { class IPluginGame; }
 
-class InstanceManager {
+class Settings;
+class PluginContainer;
 
+
+class Instance
+{
 public:
+  enum class SetupResults
+  {
+    Ok,
+    BadIni,
+    IniMissingGame,
+    PluginGone,
+    GameGone,
+    MissingVariant
+  };
+
+  Instance(QDir dir, bool portable, QString profileName={});
+
+  SetupResults setup(PluginContainer& plugins);
+
+  void setGame(const QString& name, const QString& dir);
+  void setVariant(const QString& name);
+
+  QString name() const;
+  QString gameName() const;
+  QString gameDirectory() const;
+  QDir directory() const;
+  MOBase::IPluginGame* gamePlugin() const;
+  QString profileName() const;
+  QString iniPath() const;
+  bool isPortable() const;
 
-  static InstanceManager &instance();
+private:
+  QDir m_dir;
+  bool m_portable;
+  QString m_gameName, m_gameDir, m_gameVariant;
+  MOBase::IPluginGame* m_plugin;
+  QString m_profile;
+
+  SetupResults getGamePlugin(PluginContainer& plugins);
+  void getProfile(const Settings& s);
+};
 
-  QString determineDataPath();
-  void clearCurrentInstance();
+
+class InstanceManager
+{
+public:
+  static InstanceManager &instance();
```

Comment:
Maybe rename this? Maybe `manager()`? I know it's for a singleton, but since it manages instances it can be misleading, e.g. `InstanceManager::instance().currentInstance()`.

Reply:
I agree, but it follows convention. I could rename it `singleton()` or something, but I'm not sure it's worth it.

Reply:
I would prefer a different name as well. singleton sounds good enough. I would definitely be confused when reading code.

Reply:
There's also `SingleInstance`, `InstanceManager` and `Instance`. I might rename the first one and stop using `instance` for multiple processes.

Reply:
Renamed to `singleton()`.